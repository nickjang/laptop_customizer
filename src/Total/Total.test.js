import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Total from './Total';

const selected = {
  Processor: {
    name: '17th Generation Intel Core HB (7 Core with donut spare)',
    cost: 700
  },
  'Operating System': {
    name: 'Ubuntu Linux 16.04',
    cost: 200
  },
  'Video Card': {
    name: 'Toyota Corolla 1.5v',
    cost: 1150.98
  },
  Display: {
    name: '15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs',
    cost: 1500
  }
}

describe('<Total />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Total />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders empty when no props supplied', () => {
    const wrapper = shallow(<Total />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Total with the total value from selected options', () => {
    const wrapper = shallow(<Total selected={selected} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});