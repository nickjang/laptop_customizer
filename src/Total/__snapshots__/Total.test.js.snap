// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Total /> renders Total with the total value from selected options 1`] = `
<div
  className="summary__total"
>
  <div
    className="summary__total__label"
  >
    Total
  </div>
  <div
    className="summary__total__value"
  >
    $3,550.98
  </div>
</div>
`;

exports[`<Total /> renders empty when no props supplied 1`] = `
<div
  className="summary__total"
>
  <div
    className="summary__total__label"
  >
    Total
  </div>
  <div
    className="summary__total__value"
  >
    $0.00
  </div>
</div>
`;
